sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -0,0 +1,546 @@
+	cuda? ( dev-util/nvidia-cuda-toolkit:= )
+	ffmpeg? ( media-video/ffmpeg:= )
+	gdal? ( sci-libs/gdal:= )
+	java? ( >=virtual/jdk-1.8:* )
+	json? ( dev-libs/jsoncpp:= )
+	mpi? (
+		sci-libs/h5part
+		sys-cluster/openmpi[cxx,romio]
+		virtual/mpi[cxx,romio]
+	)
+	mysql? ( virtual/mysql )
+	odbc? ( dev-db/unixODBC )
+	offscreen? ( media-libs/mesa[osmesa] )
+	pegtl? ( dev-libs/pegtl )
+	postgres? ( dev-db/postgresql:= )
+	python? ( ${PYTHON_DEPS} )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtsql:5
+		dev-qt/qtwidgets:5
+	)
+	rendering? (
+		media-libs/freeglut
+		media-libs/glew:=
+		sci-libs/proj:=
+		virtual/opengl
+		x11-libs/gl2ps
+	)
+	tbb? ( dev-cpp/tbb )
+	tk? ( dev-lang/tk:= )
+	video_cards_nvidia? ( x11-drivers/nvidia-drivers[tools,static-libs] )
+	views? (
+		x11-libs/libICE
+		x11-libs/libXext
+	)
+	web? ( ${WEBAPP_DEPEND} )
+	$(python_gen_cond_dep '
+		python? (
+			all-modules? ( dev-python/mpi4py[${PYTHON_MULTI_USEDEP}] )
+			boost? ( dev-libs/boost:=[mpi?,python?,${PYTHON_MULTI_USEDEP}] )
+			gdal? ( sci-libs/gdal:=[python?,${PYTHON_MULTI_USEDEP}] )
+			mpi? ( dev-python/mpi4py[${PYTHON_MULTI_USEDEP}] )
+		)
+	')
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-libs/jsoncpp:=
+	dev-libs/pegtl
+	mpi? ( app-admin/chrpath )
+	openmp? (
+		|| (
+			sys-devel/gcc[openmp(+)]
+			sys-devel/clang-runtime[openmp(+)]
+		)
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-0001-fix-kepler-compute_arch-if-CUDA-toolkit-11-is-used.patch
+	"${FILESDIR}"/${P}-0002-use-cmake-internal-auto-search-for-cuda-devices.patch
+)
+
+S="${WORKDIR}"/VTK-${PV}
+
+DOCS=( CONTRIBUTING.md README.md )
+
+pkg_pretend() {
+	if use cuda; then
+		# NOTE: This should actually equal to (number of build jobs)*7G,
+		# as any of the cuda compile tasks can take up 7G!
+		# 10.2 GiB install directory, 6.4 GiB build directory with max. USE flags
+		CHECKREQS_MEMORY="7G"
+		CHECKREQS_DISK_BUILD="17G"
+		check-reqs_pkg_setup
+	fi
+}
+
+pkg_setup() {
+	if use cuda; then
+		CHECKREQS_MEMORY="7G"
+		CHECKREQS_DISK_BUILD="17G"
+		check-reqs_pkg_setup
+	fi
+	use java && java-pkg-opt-2_pkg_setup
+	use python && python-single-r1_pkg_setup
+	use web && webapp_pkg_setup
+}
+
+src_unpack() {
+	default
+
+	if use test; then
+		ebegin "Copying extra test datafiles"
+		mkdir -p "${S}/.ExternalData/MD5" || die
+		for i in ${MISSING_TEST_FILES[@]}; do
+			cp -i "${DISTDIR}/${i}" "${S}/.ExternalData/MD5" || die
+		done
+		unset i
+		eend "$?"
+	fi
+}
+
+src_prepare() {
+	# If we have system libraries available use these and delete
+	# the respecting files in ${S}/ThirdParty to save some space.
+	# Note: freetype is omitted: vtk doesn't build with freetype-2.10.4
+	# Note: libharu is omitted: vtk needs an updated version (2.4.0)
+	# Note: no valid xdmf2 targets are found for system xdmf2
+	# Note: no valid target found for h5part and mpi4py
+	# ToDo: diy2 exodusII freetype h5part kissfft libharu verdict vpic
+	#	vtkm xdmf2 xdmf3 zfp
+	local -a DROPS=( doubleconversion eigen expat gl2ps glew hdf5 jpeg
```

Comment:
I'd probably expand this onto one per line or similar but it's okay as-is too

Reply:
It'd be a loong list, so I'd prefer the more compact layout.